shell/platform/tizen/external_texture_gl.cc

```diff
@@ -30,57 +30,89 @@ struct ExternalTextureGLState {
 
 static std::atomic_long nextTextureId = {1};
 
+static void MarkTbmSurfaceToUse(void* surface) {
+#ifndef WEARABLE_PROFILE
+  FT_ASSERT(surface);
+  tbm_surface_h tbm_surface = (tbm_surface_h)surface;
+  tbm_surface_internal_ref(tbm_surface);
+#endif
+}
+
+static void UnmarkTbmSurfaceToUse(void* surface) {
+  FT_ASSERT(surface);
+  tbm_surface_h tbm_surface = (tbm_surface_h)surface;
+#ifndef WEARABLE_PROFILE
+  tbm_surface_internal_unref(tbm_surface);
+#else
+  tbm_surface_destroy(tbm_surface);
```

Comment:
why destroy tbm surface here?

Reply:
It reflects the current implementation that does not use ref/unref.

Reply:
The current implementaion does not work tho.

Reply:
@swift-kim You are right.  
@swift-kim @xuelian-bai 
if the cost of using map/unmap is high,  I think using ref/unref is fine until find an alternative.

Reply:
maybe we can use the method to get the function pointer from libtbm.so.1

Reply:
@xuelian-bai 
> tbm_surface_internal_ref and tbm_surface_internal_unref are used to lock/unlock the tbm surface from free by mmplayer, so they can't be deleted. Except these two APIs, we can also use tbm_surface_map/tbm_surface_unmap to lock/unlock surface, but they are expensive.

I don't understand yet why you mention of map/unmap at here. Can you give me a rough code snippet regarding how to use it? (#80 is closed, so I'd like to discuss here. 😃 )

Reply:
ref is to lock tbm surface from being freed, ref function will increase ref count. Map function won't increase the ref count, but it will increase lock count, if lock count is increased, that buffer can also not be destroyed.

Reply:
> maybe we can use the method to get the function pointer from libtbm.so.1

@xuelian-bai  Well, I would appreciate it if you could be more specific.

> ref is to lock tbm surface from being freed, ref function will increase ref count. Map function won't increase the ref count, but it will increase lock count, if lock count is increased, that buffer can also not be destroyed.

@xuelian-bai Ah... I understood. I looked at the implementation too, but I was curious about what you were saying because it didn't increase the ref count. However, map/unmap still seems like a very expensive operation.

Reply:
@xuelian-bai What do you think of the approach I suggested for the #80? Then, we just use and erase packet without having to worry about refcnt?

Reply:
@xuelian-bai 

> I didn't get it. You use media_packet instead of tbm surface, but how to generate texture by using media_packet?

Basically, we have to do same work which is performed by video or camera plug-in.. Code at #80 is  just to show how change of interface will be... but the explanation seems to be insufficient. 
I'll make a PR , so let's discuss whether it applies or not.

Reply:
@xuelian-bai 
My suggestion is in number #82.

Reply:
@bwikbs, @xuelian-bai 
This PR fixes a crash and allows to use external textures in profiles other than wearable.  
So, how about that we go to #82 for the discussion on how to create an external texture?

Reply:
Could you merge this PR, if there is no issue other than things related to #82

Reply:
That's a good idea. Shall we go to #82 for discussion regarding this?